plug-ins/sample-data-import.ny

```diff
@@ -104,4 +104,4 @@ $control bad-data (_ "Invalid data handling") choice (("ThrowError" (_ "Throw Er
     (close stream)
     audio-out))
 
-(catch 'err (sound-from-file filename))
+(catch 'err (sound-from-file))
```

Comment:
why can `filename` be omitted here?

Reply:
(Nyquist is case insensitive)

> why can `filename` be omitted here?

Because `FILENAME` (previously `filename`) is a global variable, so can be read directly from the global scope. It is not being modified anywhere, so we are correctly treating it as a global constant, as indicated by the capitalisation of the name.

Previously, `filename` within the function `sound-from-file` redefined the variable name from the outer scope as a variable with local scope. This is generally considered bad practice as it can be unclear that `filename` within the  function is a different variable than `filename` from the outer scope. In the new code, it is explicit that we are using  `FILENAME` as a global constant.